### PR TITLE
[SPARK-17964][SparkR] Enable SparkR with Mesos client mode and cluster mode

### PR DESCRIPTION
**Pull request by susanxhuynh**
## What changes were proposed in this pull request?

Enabled SparkR with Mesos client mode and cluster mode. Just a few changes were required to get this working on Mesos: (1) removed the SparkR on Mesos error checks and (2) do not require "--class" to be specified for R apps. The logic to check spark.mesos.executor.home was already in there.

@sun-rui 

## How was this patch tested?

1. SparkSubmitSuite
2. On local mesos cluster (on laptop): ran SparkR shell, spark-submit client mode, and spark-submit cluster mode, with the "examples/src/main/R/dataframe.R" example application.
3. On multi-node mesos cluster: ran SparkR shell, spark-submit client mode, and spark-submit cluster mode, with the "examples/src/main/R/dataframe.R" example application. I tested with the following --conf values set: spark.mesos.executor.docker.image and spark.mesos.executor.home

This contribution is my original work and I license the work to the project under the project's open source license.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/api/r/RUtils.scala [hunk 1]
```diff
@@ -84,7 +84,7 @@ private[spark] object RUtils {
       }
     } else {
       // Otherwise, assume the package is local
-      // TODO: support this for Mesos
+      // For Mesos, the path is also under SPARK_HOME.
```
Comment (mgummelt) [inline core/src/main/scala/org/apache/spark/api/r/RUtils.scala]: This comment seems unnecessary now.  Maybe remove it?

**Review by mgummelt — APPROVED** [review]

**Comment by mgummelt** [comment]
ok to test

**Comment by mgummelt** [comment]
@srowen LGTM

File: core/src/main/scala/org/apache/spark/api/r/RUtils.scala (hunk 1 above)
Comment (susanxhuynh) [inline core/src/main/scala/org/apache/spark/api/r/RUtils.scala]: I have removed this comment.

**Comment by felixcheung** [comment]
This might not handle packages the best it can on mesos but that could be a follow-up
@sun-rui

**Comment by susanxhuynh** [comment]
@felixcheung There are separate issues for packages that are not included in this PR:
- [Support using 3rd-party R packages on Mesos](https://issues.apache.org/jira/browse/SPARK-17968)
- [Support Spark packages with R code on Mesos](https://issues.apache.org/jira/browse/SPARK-17966)

Also, I could not find any documentation about packages. It would be helpful to fill in some information about requirements or testing instructions in the links above.

**Comment by mgummelt** [comment]
retest this please

**Comment by mgummelt** [comment]
retest this please

**Comment by srowen** [comment]
Jenkins add to whitelist

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Merged to master/2.1

**Pull request closed without merging**